### ISSUE TITLE
mysql_database_user's action grant is not idempotent

### ISSUE THREAD

**Issue by kostyrevaa** [opscode-cookbooks/database#155]
```
mysql_database_user 'db_admin2' do
  connection(
    :host => '127.0.0.1',
    :username => 'root',
    :password => 'learnchef_mysql'
  )
  password 'database_password2'
  database_name 'products'
  host '127.0.0.1'
  action [:create, :grant]
end
```
always change 1 resource
```
   * mysql_database_user[db_admin2] action grant
      - Granting privs for 'db_admin2'@'127.0.0.1'
Chef Client finished, 1/116 resources updated in 5.187874261 seconds
```

**Comment by nik045** [comment]
:+1:

**Comment by someara** [comment]
Fixed up in 4.0.7
